Cython/Build/Dependencies.py

```diff
@@ -1034,7 +1034,8 @@ def copy_to_build_dir(filepath, root=os.getcwd()):
                 # setup for out of place build directory if enabled
                 if build_dir:
                     if os.path.isabs(c_file):
-                        warnings.warn("build_dir has no effect for absolute source paths")
+                        c_file = os.path.splitdrive(c_file)[1]
+                        c_file = c_file.split(os.sep, 1)[1]
```

Comment:
Hmm. Not sure right now if this should be changed. Probably not wrong, but the effect is much wider than just pyximport.

Reply:
Unfortunately, yes. I did quick grep and it seems that `build_dir` argument is of `cythonize()` is used only here:
* `Cython/Build/Inline.py` in `cython_compile()` and is passed by default compiler options from https://github.com/cython/cython/blob/a8f960e9c6ad27302b5704cb3b6a3f94a3472361/Cython/Compiler/Options.py#L714
* `Cython/Distutils/build_ext.py` - this is our change so we can "ignore" it.

My humble opinion is that building relative path from absolute one is better than just ignoring the file. Plus if this line is executed somewhere, it means that compilation of given file is ignored what will result in broken build right? 

Plus 3.0 is ideal release for introducing this kind of change.